maths/double_factorial_iterative.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * @function DoubleFactorialIterative
+ * @description Calculate the double factorial of a number (iterative implementation)
+ * @summary In mathematics, double factorial of a number n is denoted by n!!. 
+ * It is not to be confused with (n!)!, which is the factorial function iterated twice.  
+ * The double factorial is the product of all positive integers upto n that have the same parity (odd or even)
+ * as n.  
+ * Therefore,  
+ * 9!! = 9 . 7 . 5 . 3 . 1  
+ * 10!! = 10 . 8 . 6 . 4 . 2
+ * 
+ * Please note that for factorials of even numbers, the series ends at 2.
+ * @see [Wikipedia](https://en.wikipedia.org/wiki/Double_factorial)
+ * @see [Mathworld](https://mathworld.wolfram.com/DoubleFactorial.html)
+ * @see [GeeksForGeeks](https://www.geeksforgeeks.org/double-factorial/)
+ * @example DoubleFactorialIterative(4) = 8
+ * @example DoubleFactorialIterative(5) = 15
+ */
+const DoubleFactorialIterative = (n: number) => {
+  if(n < 0) throw new RangeError("The number needs to be non-negative")
+  if(n === 0) return 1;
```

Comment:
Is this needed? It seems the code should handle this properly without it.

Reply:
Oh yeah, you're right, the code would perform just fine without the if(n===0) statement, I'll remove it right away